tenants-certified-attributes-updater/src/main/scala/it/pagopa/interop/tenantscertifiedattributesupdater/util/Utils.scala

```diff
@@ -65,22 +65,31 @@ object Utils {
     attributesIndex: Map[UUID, AttributeInfo]
   ): TenantActions = {
 
+    val filteredTenant: List[PersistentTenant] = tenants.filter(_.selfcareId.isDefined)
+
     val originsFromPartyRegistry: List[String] = institutions.map(_.origin).distinct
 
     val filteredAttributesIndex: Map[UUID, AttributeInfo] =
       attributesIndex.filter(a => originsFromPartyRegistry.contains(a._2.origin))
 
-    val fromRegistry: List[TenantSeed] = institutions.filter(_.id.nonEmpty).map(createTenantSeed)
+    val tenantsMap: Map[PersistentExternalId, String] = filteredTenant.map(t => (t.externalId, t.name)).toMap
+
+    val fromRegistry: List[TenantSeed] = institutions
+      .filter(institution => institution.id.nonEmpty)
+      .collect {
+        case i @ Institution(_, originId, _, _, _, _, _, _, _, origin, _)
+            if tenantsMap.contains(PersistentExternalId(origin, originId)) =>
+          i
+      }
```

Comment:
minor but more readable
```suggestion
      .filter(institution => institution.id.nonEmpty && tenantsMap.contains(PersistentExternalId(institution.origin, institution.originId)))
```